model/user.go

```diff
@@ -861,6 +885,24 @@ func IsValidUsername(s string) bool {
 	return true
 }
 
+func IsValidUsernameAllowRemote(s string) bool {
+	if len(s) < USER_NAME_MIN_LENGTH || len(s) > USER_NAME_MAX_LENGTH {
+		return false
+	}
+
+	if !validUsernameCharsForRemote.MatchString(s) {
+		return false
+	}
+
+	for _, restrictedUsername := range restrictedUsernames {
+		if s == restrictedUsername {
+			return false
+		}
+	}
```

Comment:
If it's not too much trouble can we make `restrictedUsernames` a map and further simplify?

Reply:
Fixed via https://github.com/mattermost/mattermost-server/pull/17347/commits/0b1337e31f9e880ac1098045a7f34c9adf44ce07